### PR TITLE
Introduce parallel stage in libres' jenkinsfile

### LINKED ISSUE #1812: Set sequential tests stages in non-komodo jenkins pipeline as parallel
**Problem description.**
Currently [ctests](https://github.com/equinor/ert/blob/47718c11af96c08e04fd295dc9b7db4187e5cf02/Jenkinsfile-libres#L24), [equinor tests](https://github.com/equinor/ert/blob/47718c11af96c08e04fd295dc9b7db4187e5cf02/Jenkinsfile-libres#L29) and [normal-pytest](https://github.com/equinor/ert/blob/47718c11af96c08e04fd295dc9b7db4187e5cf02/Jenkinsfile-libres#L34)
are run sequentially, but this can be done parallelly to speed up the total run of the pipeline

**Solution**
Implement a declarative Jenkins parallel structure for the before mention stages

**Additional information**
possible starting point
```
        stage('tests') {
            failFast true
            parallel {
                stage('run ctest') {
                        steps {
                            sh 'sh libres/testjenkins.sh run_ctest'
                        }
                    }
                stage('run pytest_equinor') {
                        steps {
                            sh 'sh libres/testjenkins.sh run_pytest_equinor'
                        }
                    }
                stage('run pytest_normal') {
                        steps {
                            sh 'sh libres/testjenkins.sh run_pytest_normal'
                        }
                }
            }
        }
```

### PR DESCRIPTION
**Pull request by sregales-TNO**
**Issue**
Resolves #1812 

**Approach**
group the test stages of the pipeline together and run them in parallel

### CODE REVIEW

File: ci/jenkins/Jenkinsfile-libres [hunk 1]
```diff
@@ -10,30 +10,35 @@ pipeline {
...
-	stage('run pytest_equinor') {
-            steps {
-                sh 'sh ci/jenkins/test_libres_jenkins.sh run_pytest_equinor'
-            }
-        }
-	stage('run pytest_normal') {
-            steps {
-                sh 'sh ci/jenkins/test_libres_jenkins.sh run_pytest_normal'
+        stage('tests') {
+            failFast true
```
Comment (jondequinor) [inline ci/jenkins/Jenkinsfile-libres:25]: Just wondering what the rationale for failfast here is?

**Review by dotfloat — APPROVED** [review]

File: ci/jenkins/Jenkinsfile-libres (hunk 1 above)
Comment (sregales-TNO) [reply ci/jenkins/Jenkinsfile-libres:25]: If Ctests failed, then there is no reason to continue with the other two.

Comment (jondequinor) [reply ci/jenkins/Jenkinsfile-libres:25]: There's a lot of code in ERT that is not using any of the C bindings. So one reason to continue is that the result of those tests might be interesting.

Comment (sregales-TNO) [reply ci/jenkins/Jenkinsfile-libres:25]: okay I will remove failfast

**Review by jondequinor — APPROVED** [review]

**Pull request merged**